src/common/components/table/Table.tsx

```diff
@@ -403,48 +414,69 @@ export function Table<T>({
-                  }}
-                  minW={
-                    cell.column.columnDef.minSize ? `${cell.column.columnDef.minSize}px` : undefined
-                  }
-                  maxW={
-                    cell.column.columnDef.maxSize ? `${cell.column.columnDef.maxSize}px` : undefined
-                  }
-                  w={cell.column.columnDef.size ? `${cell.column.columnDef.size}px` : undefined}
-                  boxSizing="border-box"
-                >
-                  {flexRender(cell.column.columnDef.cell, cell.getContext())}
-                </ChakraTable.Cell>
-              );
-            })}
-          </ChakraTable.Row>
-        ))}
+        {table.getRowModel().rows.map((row, rowIndex) => {
+          const firstCell = row.getVisibleCells()[0];
+          const isSpanRow = firstCell?.column.columnDef.meta?.isSpanRow?.(row.original) ?? false;
+
+          return (
+            <ChakraTable.Row
+              key={row.id}
+              bg="transparent"
+              css={{
+                '& > td:first-of-type': {
+                  borderTopLeftRadius: 'redesign.md',
+                  borderBottomLeftRadius: 'redesign.md',
+                },
+                '& > td:last-of-type': {
+                  borderTopRightRadius: 'redesign.md',
+                  borderBottomRightRadius: 'redesign.md',
+                },
+              }}
+              className="group"
+              minH={13}
+            >
+              {row.getVisibleCells().map((cell, columnIndex) => {
+                if (isSpanRow && columnIndex > 0) {
+                  return null;
+                }
+                return (
+                  <ChakraTable.Cell
+                    key={cell.id}
+                    py={3}
+                    px={[2, 2, 2, `clamp(12px, calc(48px / ${columns.length}), 16px)`]}
+                    css={{ ...getCommonPinningStyles(cell.column) }}
+                    _groupHover={{
+                      bg: 'surfacePrimary',
+                    }}
+                    minW={
+                      cell.column.columnDef.minSize
+                        ? `${cell.column.columnDef.minSize}px`
+                        : undefined
+                    }
+                    maxW={
+                      cell.column.columnDef.maxSize
+                        ? `${cell.column.columnDef.maxSize}px`
+                        : undefined
+                    }
+                    w={
+                      cell.column.columnDef.size
+                        ? `${cell.column.columnDef.size}px`
+                        : isSpanRow
+                          ? 'auto !important'
+                          : undefined
+                    }
+                    boxSizing="border-box"
+                    textAlign={cell.column.columnDef.meta?.textAlign}
+                    colSpan={isSpanRow ? columns.length : 1}
+                    bg={isSpanRow ? 'surfacePrimary' : undefined}
```

Comment:
This seems like specific knowledge about the blocks table, which shouldn't be in the table component